frame/system/src/weights.rs

```diff
@@ -74,3 +65,399 @@ impl ExtrinsicsWeight {
+pub struct ForDispatchClass<T> {
+	/// Value for `Normal` extrinsics.
+	pub normal: T,
+	/// Value for `Operational` extrinsics.
+	pub operational: T,
+	/// Value for `Mandatory` extrinsics.
+	pub mandatory: T,
+}
+
+impl<T: Copy> ForDispatchClass<T> {
+	/// Create new `ForDispatchClass` with the same value for every class.
+	pub fn new(val: T) -> Self {
+		Self {
+			normal: val,
+			operational: val,
+			mandatory: val,
+		}
+	}
+
+	/// Set the value to given class.
+	pub(crate) fn set(&mut self, value: T, class: ExtrinsicDispatchClass) {
+		match class {
+			ExtrinsicDispatchClass::All => {
+				self.normal = value;
+				self.operational = value;
+				self.mandatory = value;
+			},
+			ExtrinsicDispatchClass::One(DispatchClass::Normal) => {
+				self.normal = value;
+			},
+			ExtrinsicDispatchClass::One(DispatchClass::Operational) => {
+				self.operational = value;
+			},
+			ExtrinsicDispatchClass::One(DispatchClass::Mandatory) => {
+				self.mandatory = value;
+			},
+		}
+	}
+
+	/// Get value for given class.
+	pub fn get(&self, class: DispatchClass) -> T {
+		match class {
+			DispatchClass::Normal => self.normal,
+			DispatchClass::Operational => self.operational,
+			DispatchClass::Mandatory => self.mandatory,
+		}
+	}
+}
+#[derive(Default, RuntimeDebug)]
+pub struct ValidationErrors {
+	pub has_errors: bool,
+	#[cfg(feature = "std")]
+	pub errors: Vec<String>,
+}
+
+#[cfg(feature = "std")]
+macro_rules! error_assert {
+	($cond : expr, $err : expr, $format : expr $(, $params: expr )*) => {
+		if !$cond {
+			$err.has_errors = true;
+			$err.errors.push(format!($format $(, &$params )*));
+		}
+	}
+}
+#[cfg(not(feature = "std"))]
+macro_rules! error_assert {
+	($cond : expr, $err : expr, $format : expr $(, $params: expr )*) => {
+		if !$cond {
+			$err.has_errors = true;
+		}
+	}
+}
+
+pub type ValidationResult = Result<BlockWeights, ValidationErrors>;
+
+/// A ratio of `Normal` dispatch class within block, used as default value for
+/// `BlockWeight` and `BlockLength`. The `Default` impls are provided mostly for convenience
+/// to use in tests.
+const DEFAULT_NORMAL_RATIO: Perbill = Perbill::from_percent(75);
+
+/// Block weight limits configuration.
+#[derive(RuntimeDebug, Clone)]
+pub struct BlockWeights {
+	/// Base weight of block execution.
+	pub base_block: Weight,
+	/// Base weight of single extrinsic of given class.
+	pub base_extrinsic: ForDispatchClass<Weight>,
+	/// Maximal weight of single extrinsic of given class. Should NOT include `base_extrinsic` cost.
+	///
+	/// `None` indicates that this class of extrinsics doesn't have a limit.
+	pub max_extrinsic: ForDispatchClass<Option<Weight>>,
+	/// Block reserved allowance for all extrinsics of a particular class.
+	///
+	/// Setting to `None` indicates that extrinsics of that class are allowed
+	/// to go over total block weight (but at most `max_allowance`).
+	/// Setting to `Some(x)` guarantees that at least `x` weight of particular class
+	/// is processed in every block.
```

Comment:
```suggestion
	/// Setting to `Some(x)` guarantees that at least `x` weight of particular class
	/// is not processed in every block.
```

or am I misreading it?